### ISSUE TITLE
Cannot evaluate sizeof arrays on compiletime

### ISSUE THREAD

**Issue by zefciu** [nim-lang/Nim#5802]
The size of array is known on compile time, yet nim is unable to evaluate `sizeof(SomeArrayType)`. This limits the ability to use computations on compile time. Found this trying to compute the md5 raindbow table on compile time.

Minimal example:
```
type                                                                            
  SomeType = array[0..10, string]                                               
                                                                                
const some_type_size = sizeof(SomeType)
```

Result:
```
const.nim(13, 30) Error: cannot evaluate 'sizeof(SomeType)'
```

**Comment by markus-oberhumer** [comment]
Maybe somewhat releated, `len(SomeArrayType)` also does not compile.

```
var  v:  array[1..10, int]
type T = array[1..10, int]

var   a: int = v.sizeof
var   b: int = T.sizeof
const c: int = v.sizeof   # does not compile
const d: int = T.sizeof   # does not compile

var   e: int = v.len
var   f: int = T.len      # does not compile
const g: int = v.len
const h: int = T.len      # does not compile
```

**Comment by krux02** [comment]
Well I have a pull request open to make sizeof work on all types at compile time very well here:
https://github.com/nim-lang/Nim/pull/5664
But my checks are failing, because the packed pragma is not correctly implemented:
https://github.com/nim-lang/Nim/issues/5824

**Comment by markus-oberhumer** [comment]
It looks that `len(SomeArrayType)` has been implemented by #6032.

**Comment by Araq** [comment]
This is now possible.
